compiler/parser.nim

```diff
@@ -914,7 +914,8 @@ proc simpleExprAux(p: var Parser, limit: int, mode: PrimaryMode): PNode =
     pragmaExp.add result
     pragmaExp.add p.parsePragma
     result = pragmaExp
-  result = parseOperators(p, result, limit, mode)
+  if not (mode == pmTypeDef and result.kind in {nkObjectTy, nkTupleTy}):
+    result = parseOperators(p, result, limit, mode)
```

Comment:
Pre-existing but this `mode` parsing is not expressible via the grammar and hence very bad.